src/drivers/intel/ssp/ssp.c

```diff
@@ -98,10 +119,10 @@ static void ssp_empty_rx_fifo(struct dai *dai)
 		dai_dbg(dai, "ssp_empty_rx_fifo(), before flushing, entries %d", entries);
 
 		/* let DMA consume data or read RX FIFO directly */
-		if (direct_reads) {
-			for (i = 0; i < entries + 1; i++)
-				ssp_read(dai, SSDR);
-		}
+#if CONFIG_DMA_SUSPEND_DRAIN
+		for (i = 0; i < entries + 1; i++)
+			ssp_read(dai, SSDR);
+#endif
```

Comment:
I'm not certain about this. Even if the DMA is active (not disabled) but the RX FIFO does not have trigger amount of data then the DMA will not clear the FIFO, so we just spin w/o any reason.

Reply:
@ujfalusi Ack, so this will not drain the samples that fall below the DMA trigger threshold and the function will timeout (which at 8kHz is a long timeout). OTOH, in old code fifo is emptied directly and DMA is left without trailing data, so not great either.